### PR TITLE
add github actions

### PR DESCRIPTION
**Pull request by knorth55**
add github actions for indigo, kinetic and melodic.
- exclude `instance_occlsegm` and `grasp_fusion` test because of gdrive download maximum trials

### CODE REVIEW

**Comment by knorth55** [comment]
all test passed!

**Comment by knorth55** [comment]
@pazeshun I add github actions for faster CI than travis CI.
we do not run several tests in github actions, and we run whole tests in travis CI.
what do you think?

Another issue is that we want to support noetic for the near future.
I started doing the integration in #2746, but we have to wait to upstream repositories.

**Comment by pazeshun** [comment]
> @pazeshun I add github actions for faster CI than travis CI.
> we do not run several tests in github actions, and we run whole tests in travis CI.
> what do you think?

I think it's OK.
Thank you very much for heavy installation!!!

>     * exclude `instance_occlsegm` and `grasp_fusion` test because of gdrive download maximum trials

Just out of curiosity, is this actually avoidable by using cache?
https://dev.to/github/caching-dependencies-to-speed-up-workflows-in-github-actions-3efl
Anyway, we currently have no time to use this, and travis targeting master branch is enough, I think.

File: .github/workflows/indigo.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  indigo:
+    runs-on: ubuntu-latest
+    name: indigo
+
+    container: jskrobotics/ros-ubuntu:14.04
+
+    steps:
+      - name: Install latest git ( use sudo for ros-ubuntu, remove sudo for ubntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkkout .travis submodules
```
Comment (pazeshun) [inline .github/workflows/indigo.yaml:12]:
```suggestion
      - name: Install latest git ( use sudo for ros-ubuntu, remove sudo for ubntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkout .travis submodules
```

**Review by pazeshun — CHANGES_REQUESTED** [review]

File: .github/workflows/indigo.yaml (hunk 1 above)
Comment (pazeshun) [reply .github/workflows/indigo.yaml:12]: Oh, I understand this is copied from jsk_travis.
Perhaps we should leave it as it is.
I'll follow @knorth55 's thought.

**Comment by knorth55** [comment]
@pazeshun thanks.

>Just out of curiosity, is this actually avoidable by using cache?
https://dev.to/github/caching-dependencies-to-speed-up-workflows-in-github-actions-3efl
Anyway, we currently have no time to use this, and travis targeting master branch is enough, I think.

I will take a look later.

File: .github/workflows/indigo.yaml (hunk 1 above)
Comment (pazeshun) [reply .github/workflows/indigo.yaml]: @knorth55 コメントが間に合わず申し訳ないんですが、前のコメントの通り、typoそのままでもいいかなと思っています。

Comment (knorth55) [reply .github/workflows/indigo.yaml]: typoだしなおしておきましょう

File: .github/workflows/indigo.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+jobs:
+  indigo:
+    runs-on: ubuntu-latest
+    name: indigo
+
+    container: jskrobotics/ros-ubuntu:14.04
+
+    steps:
+      - name: Install latest git ( use sudo for ros-ubuntu, remove sudo for ubntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkout .travis submodules
```
Comment (pazeshun) [inline .github/workflows/indigo.yaml]:
```suggestion
      - name: Install latest git (use sudo for ros-ubuntu, remove sudo for ubuntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkout .travis submodules
```

File: .github/workflows/indigo.yaml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        run: sudo apt-get update && sudo apt-get install -y software-properties-common && sudo apt-get update && sudo add-apt-repository -y ppa:git-core/ppa && sudo apt-get update && sudo apt-get install -y git
+      - name: Before Checkout # need for actoins/checkout with ros-ubuntu container
```
Comment (pazeshun) [inline .github/workflows/indigo.yaml]:
```suggestion
      - name: Before Checkout # need for actions/checkout with ros-ubuntu container
```

File: .github/workflows/kinetic.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  kinetic:
+    runs-on: ubuntu-latest
+    name: kinetic
+
+    container: jskrobotics/ros-ubuntu:16.04
+
+    steps:
+      - name: Install latest git ( use sudo for ros-ubuntu, remove sudo for ubntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkkout .travis submodules
```
Comment (pazeshun) [inline .github/workflows/kinetic.yml]:
```suggestion
      - name: Install latest git (use sudo for ros-ubuntu, remove sudo for ubuntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkout .travis submodules
```

File: .github/workflows/kinetic.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        run: sudo apt-get update && sudo apt-get install -y software-properties-common && sudo apt-get update && sudo add-apt-repository -y ppa:git-core/ppa && sudo apt-get update && sudo apt-get install -y git
+      - name: Before Checkout # need for actoins/checkout with ros-ubuntu container
```
Comment (pazeshun) [inline .github/workflows/kinetic.yml]:
```suggestion
      - name: Before Checkout # need for actions/checkout with ros-ubuntu container
```

File: .github/workflows/melodic.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  melodic:
+    runs-on: ubuntu-latest
+    name: melodic
+
+    container: jskrobotics/ros-ubuntu:18.04
+
+    steps:
+      - name: Install latest git ( use sudo for ros-ubuntu, remove sudo for ubntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkkout .travis submodules
```
Comment (pazeshun) [inline .github/workflows/melodic.yml]:
```suggestion
      - name: Install latest git (use sudo for ros-ubuntu, remove sudo for ubuntu container), checkout@v2 uses REST API for git<2.18, which removes .git folder and does not checkout .travis submodules
```

File: .github/workflows/melodic.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        run: sudo apt-get update && sudo apt-get install -y software-properties-common && sudo apt-get update && sudo add-apt-repository -y ppa:git-core/ppa && sudo apt-get update && sudo apt-get install -y git
+      - name: Before Checkout # need for actoins/checkout with ros-ubuntu container
```
Comment (pazeshun) [inline .github/workflows/melodic.yml]:
```suggestion
      - name: Before Checkout # need for actions/checkout with ros-ubuntu container
```

**Review by pazeshun — CHANGES_REQUESTED** [review]

File: .github/workflows/indigo.yaml (hunk 1 above)
Comment (pazeshun) [reply .github/workflows/indigo.yaml]: 了解です、他にもtypo見つけてしまったのでreviewしておきました。
jsk_travis本家にもPR送っておきます。

Comment (pazeshun) [reply .github/workflows/indigo.yaml]: > jsk_travis本家にもPR送っておきます。

よく見たら既にPR出ていました。
https://github.com/jsk-ros-pkg/jsk_travis/pull/430

**Comment by knorth55** [comment]
@pazeshun I fixed the typos before your suggestions.

**Comment by pazeshun** [comment]
> @pazeshun I fixed the typos before your suggestions.

Oops, I overlooked this. Sorry...

**Comment by knorth55** [comment]
github action cache does not work properly because of permission error.

https://github.com/ros-tooling/setup-ros-docker/issues/7
https://github.com/actions/cache/issues/404

**Comment by knorth55** [comment]
I comment out github actions cache

**Review by pazeshun — APPROVED** [review]
Thank you very much!!!

**Pull request merged**